lib/wlib/comm/PacketConstructor.cpp

```diff
@@ -0,0 +1,86 @@
+/** @file PacketConstructor.c
+ *  @brief Creates a packet for sending data from the slave systems to the main computer.
+ *
+ *  Implements the functions defined in PacketConstructor.h.
+ *
+ *  @author Heather D'Souza (heathkd)
+ *  @date November 7, 2017
+ *  @bug No known bugs.
+ */
+
+#include <iostream>
+#include "../stl/Bitset.h"
+#include "PacketConstructor.h"
+
+PacketConstructor::PacketConstructor() {
+    m_pdata[0] = 0;
+    m_pdata[1] = 0;
+    m_pdata[2] = 0;
+}
+
+PacketConstructor::~PacketConstructor() {
+    delete m_pdata;
+}
+
+PacketConstructor::PacketConstructor(float* p_data) {
+    m_pdata = p_data;
+    setStartAndEnd();
+    dataToBin();
+}
+
+float* PacketConstructor::getData() {
+    return m_pdata;
+}
+
+Bitset<64> PacketConstructor::getBitArray() {
+    return m_bitArray;
+}
+
+void PacketConstructor::setStartAndEnd() {
+    m_bitArray.set(0);
+    m_bitArray.set(63);
+}
+
+void PacketConstructor::dataToBin()
+{
+    int intPart, floatPart;
+    int posArray [] = {10,20,28,38,46,56};
+
+    for (int j=0;j<3;j++) {
+        //negative or positive bit
+        if (m_pdata[j] < 0) {
+            m_bitArray.set(posArray[j*2] - 1);
+            m_pdata[j] = m_pdata[j] * (-1);
+        }
+        else {
+            m_bitArray.reset(posArray[j*2]-1);
+        }
+
+        //separate integer and decimal part of float;
+        intPart = static_cast<int>(m_pdata[j]);
```

Comment:
We are using 16 but integers instead of the standard 32 bit `int`

Reply:
``int`` is a system dependent so it is better to use a size with the ``int``

Reply:
`uint16_t`